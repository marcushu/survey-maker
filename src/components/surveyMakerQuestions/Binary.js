import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './questionInsert.css';

export default function Binary( { addQuestion }) {
	const [question, setquestion] = useState("");
	const [positiveValue, setpositiveValue] = useState("Yes");
	const [negativeValue, setnegativeValue] = useState("No");

	const handleAddQuestion = () => {
		addQuestion({
			questionType: "BINARY",
			question: question,
			positiveIdentity: positiveValue.trim().replace(' ', '_'),
			negativeIdentity: negativeValue.trim().replace(' ', '_'),
		});

		setquestion("");
	}

	return (
		<span>
			<input type="text"
				value={question}
				onChange={e => setquestion(e.target.value)}>
			</input>
			&nbsp;
			<input type="text"
				className="yesNoTxt"
				value={positiveValue}
				onChange={e => setpositiveValue(e.target.value)}>
			</input>
			&nbsp;
			<input type="text"
				className="yesNoTxt"
				value={negativeValue}
				onChange={e => setnegativeValue(e.target.value)}>
			</input>
			&nbsp;
			<button className="add" onClick={handleAddQuestion}>+</button>
		</span>
	)
}

Binary.propTypes = {
	addQuestion: PropTypes.func
};