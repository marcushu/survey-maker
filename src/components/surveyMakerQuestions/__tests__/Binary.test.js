import React from 'react';
import {  render } from '@testing-library/react';
import Binary from '../Binary';

describe("binary question maker", () => {

    const mockfunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(<Binary addQuestion={mockfunc}/>);

        expect(asFragment()).toMatchSnapshot();
    })

    it('sets each of the two available default values', () => {
        const { getAllByRole } = render(<Binary addQuestion={mockfunc}/>);

        expect(getAllByRole("textbox")[1]).toHaveValue("Yes");
        expect(getAllByRole("textbox")[2]).toHaveValue("No");
    })
})