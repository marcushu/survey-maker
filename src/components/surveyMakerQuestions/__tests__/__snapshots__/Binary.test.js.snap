// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary question maker renders 1`] = `
<DocumentFragment>
  <span>
    <input
      type="text"
      value=""
    />
     
    <input
      class="yesNoTxt"
      type="text"
      value="Yes"
    />
     
    <input
      class="yesNoTxt"
      type="text"
      value="No"
    />
     
    <button
      class="add"
    >
      +
    </button>
  </span>
</DocumentFragment>
`;
